src/state/wallets/sagas.ts

```diff
@@ -178,7 +178,18 @@ export function* sendTransactionSaga(action: SendTransactionAction | unknown) {
 
     if (result?.code === 1) {
       const message = result.message.split('\n');
-      throw new Error(`${message[0]}: ${message[2]}`);
+      const generalMsg = message[0];
+      const detailedMsg = message[2];
+      let errorMsg = `${generalMsg}: ${detailedMsg}`;
+
+      if (detailedMsg.includes(messages.txnMempoolConflictCode18)) {
+        errorMsg = i18n.send.error.doubleSpentFunds;
+      }
+      if (detailedMsg.includes(messages.missingInputs)) {
+        errorMsg = i18n.send.error.notExistingFunds;
+      }
+
+      throw new Error(errorMsg);
```

Comment:
I'd move this logic one level lower, to `BlueElectrum.broadcast`. And instead of throwing just Error, I'd create custom types of errors like `DoubleSpentFundsError` , `NotExistingFundsError` - this way we can move some logic there and keep the code cleaner. I mean something like:
```
module.exports.broadcast = async function(hex) {
  if (!mainClient) throw new Error('Electrum client is not connected');

  try {
    return await mainClient.blockchainTransaction_broadcast(hex);
  } catch (error) {
    if (error.message.includes((messages.txnMempoolConflictCode18) {
       throw new DoubleSpentFundsError();
    }

    (...)
  }
};
```

Reply:
good idea done